### PR TITLE
[PM-11899] - send text details component

### PR DESCRIPTION
**Pull request by jaasen-livefront**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-11899

## 📔 Objective

This PR adds the send text details component with a corresponding send details and base send component to allow shared use of the logic and structure for the upcoming [send file details component](https://bitwarden.atlassian.net/browse/PM-11903). This is one of three PRs for the new create send UI as part of the extension refresh. Credit goes to @djsmith85 for helping set this up.

## 📸 Screenshots

|New|Edit|
---|---
|Screenshot 2024-09-11 at 3 04 35 PM|Screenshot 2024-09-11 at 3 04 07 PM|

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by djsmith85 — CHANGES_REQUESTED** [review]
Only a partial review

As discussed I'm uncertain about the changes to the send and sendApi services, which we should likely revert and focus this PR on the UI changes. If it breaks saving Sends on the feature-flagged part, we should fix it in a separate PR.

**Comment by jaasen-livefront** [comment]
> Only a partial review
> 
> As discussed I'm uncertain about the changes to the send and sendApi services, which we should likely revert and focus this PR on the UI changes. If it breaks saving Sends on the feature-flagged part, we should fix it in a separate PR.

Sounds good. The type changes were only a narrowing of types so reverting back to `any` won't cause any type issues.

File: apps/browser/src/_locales/en/messages.json [hunk 1]
```diff
@@ -2225,13 +2229,40 @@
...
     "message": "File"
   },
   "allSends": {
     "message": "All Sends",
     "description": "'Send' is a noun and the name of a feature called 'Bitwarden Send'. It should not be translated."
   },
+  "textHiddenByDefault": {
+    "message": "When accessing the Send, hide the text by default",
+    "description": "'Send' is a noun and the name of a feature called 'Bitwarden Send'. It should not be translated."
+  },
```
Comment (djsmith85) [inline apps/browser/src/_locales/en/messages.json:2245]: ⚠️ Message key is not used anywhere and is duplicated by `hideTextByDefault` which is used.

File: apps/browser/src/_locales/en/messages.json [hunk 2]
```diff
@@ -2225,13 +2229,40 @@
...
+  "limitSendViewsHintWithCount": {
+    "message": "No one can view this Send after the limit is reached. $ACCESSCOUNT$ views left",
+    "description": "Displayed under the limit views field on Send",
+    "placeholders": {
+      "accessCount": {
+        "content": "$1",
+        "example": "2"
+      }
+    }
+  },
```
Comment (djsmith85) [inline apps/browser/src/_locales/en/messages.json:2265]: ⚠️ These message keys are not used anywhere. Need to be removed. Likely will be introduced with the send-options component

File: libs/tools/send/send-ui/src/send-form/components/send-details/send-text-details.component.html [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<bit-section [formGroup]="sendTextDetailsForm">
+  <bit-form-field>
+    <bit-label>{{ "sendTypeTextToShare" | i18n }}</bit-label>
+    <textarea bitInput id="text" rows="6" formControlName="text"></textarea>
+    <bit-hint>{{ "sendTextDesc" | i18n }}</bit-hint>
```
Comment (djsmith85) [inline libs/tools/send/send-ui/src/send-form/components/send-details/send-text-details.component.html:5]: ⚠️ The hint is not included within the latest design.

**Review by djsmith85 — CHANGES_REQUESTED** [review]

File: libs/tools/send/send-ui/src/send-form/components/send-details/send-text-details.component.html (hunk 3 above)
Comment (djsmith85) [reply libs/tools/send/send-ui/src/send-form/components/send-details/send-text-details.component.html]: Addressed with 6c7ad7ccdc0de93910bec288ad3b977da00bc000

File: apps/browser/src/_locales/en/messages.json (hunk 2 above)
Comment (djsmith85) [reply apps/browser/src/_locales/en/messages.json]: Addressed with efffcc6cfb41b7cce7f5a59c12d6137cd465318a

**Review by djsmith85 — APPROVED** [review]
@jaasen-livefront  Addressed some minor issues myself and this is now ready to merge.

**Pull request merged**